stream-chat-android-offline/src/main/java/io/getstream/chat/android/livedata/controller/ChannelControllerImpl.kt

```diff
@@ -1017,20 +1044,39 @@ internal class ChannelControllerImpl(
 
     fun upsertMember(member: Member) = upsertMembers(listOf(member))
 
-    fun updateReads(
-        reads: List<ChannelUserRead>
-    ) {
+    fun updateReads(reads: List<ChannelUserRead>) {
         val currentUserId = domainImpl.currentUser.id
-        val copy = _reads.value ?: mutableMapOf()
-        val readMap = reads.associateBy { it.getUserId() }
-
-        // handle the current user
-        val currentUserRead = readMap[currentUserId]
-        currentUserRead?.let {
-            _read.postValue(it)
-            println("updateReads for current user to ${it.lastRead} on channel $cid")
+        val previousUserIdToReadMap = _reads.value ?: mapOf()
+        val incomingUserIdToReadMap = reads.associateBy(ChannelUserRead::getUserId).toMutableMap()
+
+        /*
+        It's possible that the data coming back from the online channel query has a last read date that's before
+        what we've last pushed to the UI. We want to ignore this, as it will cause an unread state to show in the
+        channel list.
+         */
```

Comment:
@carterhudson i don't get this, we don't show your own unread state...

Reply:
do we perhaps receive mark read events out of order? @tbarbugli is that possible?

Reply:
Hmm It's possible I misunderstood this. I give this another look.